pkg/tnf/pkg/config/cluster.go

```diff
@@ -53,12 +54,39 @@ func GetClusterConfig(ctx context.Context, kubeClient kubernetes.Interface) (Clu
 	return clusterCfg, nil
 }
 
+// getInternalIP returns the internal ip address of the node.
+// We prefer ipv6 over ipv4 during dualstack deployments.
 func getInternalIP(addresses []corev1.NodeAddress) string {
+
+	// If address list is empty, return empty string as fallback.
+	if len(addresses) == 0 {
+		return ""
+	}
+
+	ipv6Address := ""
+	ipv4Address := ""
 	for _, addr := range addresses {
-		if addr.Type == corev1.NodeInternalIP {
-			return addr.Address
+		switch addr.Type {
+		case corev1.NodeInternalIP:
```

Comment:
note that you are now picking the very last of the addresses, whereas previously you would always take the first occurrence. 

just fyi, there are always ordering issues around those IP selection functions, because people rely on the ordering of their netlink interfaces and then suddenly installations fail because we pick other IPs than before. Probably not a big concern in TNF, but I wanted to let you know.

eg. last one was https://github.com/openshift/cluster-etcd-operator/commit/c94e1c2a8ba535c2efd0cb9b73f7df5926f9274c

Reply:
That's a great catch, thanks @tjungblu , after going back and forth, I think we'll just keep this more or less the same. Seeing as the first IP gets us what we want regardless of ipv4/ipv6.